File: acmeshop/acmeshop/src/main/java/org/acme/shop/startup/InitialData.java
package org.acme.shop.startup;

import io.quarkus.hibernate.reactive.panache.common.WithTransaction;
import io.quarkus.runtime.StartupEvent;
import org.acme.shop.model.Customer;
import org.acme.shop.model.ShopOrder;
import org.acme.shop.model.Product;

import jakarta.enterprise.event.Observes;
import jakarta.transaction.Transactional;

public class InitialData {

//    @WithTransaction
//    public void initialData(@Observes StartupEvent evt) {
//        Product soap = new Product();
//        soap.setName("Soap");
//        soap.persistAndFlush().s;
//
//        Customer alice = new Customer();
//        alice.setName("Alice");
//        alice.persistAndFlush();
//
//        ShopOrder alicesOrder = new ShopOrder();
//        alicesOrder.addItem(3, soap);
//        alice.addOrder(alicesOrder);
//        alicesOrder.persist();
//    }
}

File: acmeshop/acmeshop-kotlin/src/main/kotlin/shop/startup/InitialData.kt
package shop.startup

import io.quarkus.runtime.StartupEvent
import jakarta.enterprise.event.Observes
import jakarta.transaction.Transactional
import shop.model.Customer
import shop.model.Product
import shop.model.ShopOrder

class InitialData {
    @Transactional
    fun initialData(@Observes evt: StartupEvent?) {
        val soap = Product()
        soap.name = "Soap"
        soap.persistAndFlush()
        val alice = Customer()
        alice.name = "Alice"
        alice.persistAndFlush()
        val alicesOrder = ShopOrder()
        alicesOrder.addItem(3, soap)
        alice.addOrder(alicesOrder)
        alicesOrder.persist()
    }
}